INSTALL.md: document that uthash development files are required

The following recent integration:
#967 More tpmclient cleanups
added the requirement that uthash development files be installed, but did not document this fact.
If the undocumented package is missing, the following error occurs:
```
In file included from test/integration/session-util.c:27:0:
test/integration/session-util.h:31:20: fatal error: uthash.h: No such file or directory
```

I will work on a fix to document the new package requirement from #967.

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -24,6 +24,7 @@ The following are dependencies only required when building test suites.
 * Code coverage analysis:
 * lcov
 * autoconf-archives
+* uthash development libraries and header files
 
 Most users will not need to install these dependencies.
 

